### ISSUE TITLE
Can't open filemanger to choose the boot.img

### ISSUE THREAD

**Issue by gaohuadong** [topjohnwu/Magisk#8570]
install Magisk app and launch app,  can't open the filemanger to choose the boot.img when I want to patch boot.img and click "Select and Patch a file". the platform has the filemanger (com.android.documentsui).
Does the app support Car platform?  
I have the userdebug version android image, Can I user other method to patch boot.img or install magisk ? 

Device: AOSP on Intel Platform
Android version: android 14 (Car platform)
Magisk version name: v28 or v27
Magisk version code: 28001 or 27000

<img width="829" alt="image" src="https://github.com/user-attachments/assets/cd6a08b2-b7f4-4ea0-a167-51e97bb60ee6">

**Comment by vvb2060** [comment]
Any file manager can be used, not only documentsui

**Comment by gaohuadong** [comment]
> Any file manager can be used, not only documentsui
I install some apps about file manger, example "ES 文件夹浏览器", but no usefully. it seems Car platform don't support file manager permission.

**Comment by 88fingerslukee** [comment]
I'm having the same issue with Android TV. I'm using LineageOS 22 (android 15) on an Nvidia Shield TV Pro 2019.  The app just crashes when I choose Select and Patch a file.

**Comment by yujincheng08** [comment]
Try to manually patch boot instead of using magisk app?

**Comment by gaohuadong** [comment]
> Try to manually patch boot instead of using magisk app?

how patch boot manually?

**Comment by yujincheng08** [comment]
use this script:
https://github.com/topjohnwu/Magisk/blob/master/scripts/boot_patch.sh
you can find this script in magisk's apk

**Comment by canyie** [comment]
Yeah, DocumentsUI app is missing on the Car and TV platform (there was a minimal version that aims to work on TV, not sure why it was abandoned). I think you can install the app manually as a workaround? The apk is available here: https://www.reddit.com/r/AndroidQuestions/comments/t72qac/abscent_filesdocumentsui_action_open_document/

**Comment by vvb2060** [comment]
https://issuetracker.google.com/issues/71327396#comment5
This is expected.

**Comment by vvb2060** [comment]
Start of requirements added in Android 16
Automotive device implementations:

[3.2.3.1/A-0-2] MUST have an app that handles the ACTION_GET_CONTENT, ACTION_OPEN_DOCUMENT, ACTION_OPEN_DOCUMENT_TREE, and ACTION_CREATE_DOCUMENT intents as described in the SDK documents, and provide the user affordance to access the document provider data by using DocumentsProvider API.
End of requirements added (3.2.3.1/A-0-2)

https://source.android.com/docs/compatibility/16/android-16-cdd#253_software
